### PR TITLE
 Hometask from 27.04 - updating and deleting task

### PR DESCRIPTION
**Pull request by NamiDorama**

### CODE REVIEW

File: src/pages/TaskListTab/TaskListTab.js [hunk 1]
```diff
@@ -1,23 +1,51 @@
...
   };
 
+  createNewTask = (day) => {
+    this.props.history.push(`tasks/new_task?day=${day}`);
+  };
+
+  changeTaskStatus = (key, task, day) => {
+    let tasksInWeek = [...this.state.tasksInWeek];
+
+    if (key === 'completed') {
```
Comment (iliatcymbal) [inline src/pages/TaskListTab/TaskListTab.js]: And where is _in-progress_ logic?

File: src/pages/TaskListTab/TaskListTab.js [hunk 2]
```diff
@@ -1,23 +1,51 @@
...
+      task.done = true;
+      updateTask(task)
+        .then(data => this.setState({ tasksInWeek }));
```
Comment (iliatcymbal) [inline src/pages/TaskListTab/TaskListTab.js]: If you don't use _data_ argument, do not define it

File: src/pages/TaskListTab/TaskListTab.js [hunk 3]
```diff
@@ -1,23 +1,51 @@
...
-    days: ['Вс', 'Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб'],
     tasksInWeek: null
   };
 
-
   componentDidMount() {
     getTasksList()
       .then(data => {
         this.setState({ tasksInWeek: data })
       });
```
Comment (iliatcymbal) [inline src/pages/TaskListTab/TaskListTab.js]: can be simplified
```
.then(tasksInWeek => this.setState({ tasksInWeek }));
```

File: src/pages/TaskListTab/TaskListTab.js [hunk 4]
```diff
@@ -1,23 +1,51 @@
...
+  deleteTask
+} from '../../services';
+import { days, icons } from '../../consts';
+
+import './taskListTab.scss';
 
 export class TaskListTab extends Component {
   state = {
-    days: ['Вс', 'Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб'],
     tasksInWeek: null
```
Comment (iliatcymbal) [inline src/pages/TaskListTab/TaskListTab.js]: I would set it as an empty array (then don't need to check in _render_ whether `tasksInWeek` exists)

**Review by iliatcymbal — COMMENTED** [review]
Please, see my notes. Let me know when will be fixed

File: src/pages/TaskListTab/TaskListTab.js [hunk 5]
```diff
@@ -33,21 +35,19 @@ export class TaskListTab extends Component {
     }
 
     if (key === 'in-progress') {
-      this.state.tasksInWeek[day].forEach(el => {
-        if (el.done === 'in-progress') {
-          task.done = task.done;
-        }
+      if (todayTasks.every((dayTask) => dayTask.done !== 'in-progress')) {
```
Comment (iliatcymbal) [inline src/pages/TaskListTab/TaskListTab.js]: why do you need this checking? Just make `task.done = false`

File: src/pages/TaskListTab/TaskListTab.js [hunk 6]
```diff
@@ -33,21 +35,19 @@ export class TaskListTab extends Component {
...
         task.done = 'in-progress';
```
Comment (iliatcymbal) [inline src/pages/TaskListTab/TaskListTab.js]: "done" it is a flag which can have _undefined_, _false_ or _true_ values

**Review by iliatcymbal — APPROVED** [review]
Please, fix comments and emrge into the master

File: src/pages/TaskListTab/TaskListTab.js (hunk 5 above)
Comment (NamiDorama) [reply src/pages/TaskListTab/TaskListTab.js]: Because only one task can be in-progress status, so I'm checking all tasks of the current day

Comment (iliatcymbal) [reply src/pages/TaskListTab/TaskListTab.js]: Ah, I forgot about this requirement. But then, according to spec, no more than two

**Pull request merged**